src/github.com/matrix-org/dendrite/clientapi/threepid/invites.go

```diff
@@ -66,6 +66,23 @@ type idServerStoreInviteResponse struct {
 	PublicKeys  []common.PublicKey `json:"public_keys"`
 }
 
+// ProcessIDServerError differentiate errors caused by the absence of the identity
+// server in the list of trusted servers in the configuration file from normal
+// errors caused by a failure in a process.
+// Returns the representation of a 400 error in the former case (without logging),
+// and the one of a 500 error in the latter (with logging of the error).
+func ProcessIDServerError(req *http.Request, err error) util.JSONResponse {
+	// Check whether this is an
+	mErr, ok := err.(*jsonerror.MatrixError)
+	if !ok || mErr.ErrCode != "M_SERVER_NOT_TRUSTED" {
```

Comment:
I think this check would be a lot cleaner if we used a custom error type rather than jsonerror.MatrixError here.